api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java

```diff
@@ -132,14 +133,24 @@ public FireCloudServiceImpl(
    * @return
    */
   public ApiClient getApiClientWithImpersonation(String userEmail) throws IOException {
-    // Load credentials for the firecloud-admin Service Account. This account has been granted
-    // domain-wide delegation for the OAuth scopes required by FireCloud.
-    GoogleCredentials impersonatedUserCredentials =
-        ServiceAccounts.getImpersonatedCredentials(
-            fcAdminCredsProvider.get(), userEmail, FIRECLOUD_API_OAUTH_SCOPES);
+    OAuth2Credentials delegatedCreds;
+    if (configProvider.get().featureFlags.useKeylessDelegatedCredentials) {
+      delegatedCreds =
+          new DelegatedUserCredentials(
+              ServiceAccounts.getServiceAccountEmail("firecloud-admin", configProvider.get()),
```

Comment:
I feel like the user name should be configured in the JSON, even though we don't expect it to change. IOW someone else who finds this class on github would have to change that, so it doesn't feel like it should live in code.